Documentation/SharedComponentVersions.md

```diff
@@ -0,0 +1,96 @@
+
+- **Intuitive Organization** (admittedly in the eyes of the beholder)
+
+  - Easily associate a version of the shared components with a version of the specification.
+
+  - Keep documents ancillary to the specifications with this specification such that versions are held between them.  For instance, the *ExampleComplianceDocument* needs to be tightly associated with a specific specification version. Generally, duplicating it is better than attemptiong to manage independent versions (although some organizations could avoid either).
+
+  - Need to independently manage shared components and standards for numerous standards.  For instance, .NET, Python, Ruby standards and shared components need to fit into the repo.
+
+## Proposed Organization
+
+This organization is based on tags and directories.  It makes no organizational use of branches for versions of either specs or shared components.
+
+The basic approach here is:
+
+- Create a root directory for each specification type
+- That is followed by a directory for each RELEASED version (e.g., 1.0)
+- That is followed by directories for Spec and Code (Shared Components)
+- These directories contain whatever is needed.
+- Use tags to identify editorial updates to specs
+- Use tags to identify updates to shared components
+
+Some basic attributes of this organization:
+
+- The repo image has all of the released specs and corresponding shared components.  This includes versions of the specs, and the family of specs.
+- Able to contain all of the potential types of specifications with independent management of versions of the specifications and their shared components.
+- Very simple, self-documenting, the basic root trunc always has the latest released.
+- Puts an image of everything in every instance of the repo, whereas a branch-based scheme would allow only acquiring the slice you are interested in.
+- Potentially appears dated to users of git for branch-based software releases.
+
+~~~
+IviDriverCore/
+    1.0/    # spec version
+        Spec/        # use tags for editorial changes (need a tag naming
+                       convention that identifies the last commit on 
+                       this editorial version of the spec, independent of 
+                       others, e.g.: 'IviDriverCoreSpec-2024-01-10')
+
+                       'main' should have the latest
+            IviDriverCore.md
+            Example.md
+            …
+        Code/       # part of the pattern, but none for Core
+    2.0/
+    3.0/
+    
+IviDriverNet/
+    1.0/  (spec version)
+        Spec/
+            IviDriverNet.md
+            Example.md
+            …
+        Code/    # tags or branches for multiple release versions
+                   actual organization of subdirectories depends on
+                   code organization convenience.
+
+                   Need to support complete build.
+    1.1/
```

Comment:
Explicitly calling out 1.1 indicates that we've decided on branching for a minor revision. Probably best to leave it undefined so we can evaluate and decide when we have our first need to create a minor version. Gives us more room to operate.

Reply:
This structure is just to illustrate the  intent, so I thought having a minor update in the illustration helped to clarify.

However this would not be a branch in terms of git.  If you mean a branch in terms of deploying a minor standard, yes, the general practice of the consortium is that when there is a minor change to the standards we issue a new number, and archive the preceding standard.   Current archives are in SharePoint, we could discuss if we retain them on git or move them to SharePoint --- this structure anticipates that we will archive them in git, along with the associated shared components (which can themselves still receive bug fixes).